test/compilable/testclidflags.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+set -x
+
+dir=${RESULTS_DIR}${SEP}compilable
+output_file=${dir}/testclidflags.sh.out
+
+unset DFLAGS
+
+# Force DMD to print the -v menu by passing an invalid object file
+# It will fail with "no object files to link", but print the log
+( ! "$DMD" -v foo.a 2> /dev/null ) | grep -q "dflags    (none)"
+( ! DFLAGS="-O -D" "$DMD" -v foo.a 2> /dev/null ) | grep -q "dflags    -O -D"
```

Comment:
Imho it sucks that we have to workaround DMD so hard just to get it's config output. Shouldn't we make DMD's raw config settings easier to access? Most of this will be solved by the improved Json output, but there are still many cases where just looking what flags, args and precompiled versions DMD's as helps.
Ideas:
- default for `-v` (instead of showing the help page)
- `--version=detailed`